cou/steps/__init__.py

```diff
@@ -236,3 +244,82 @@ async def run(self) -> Any:
             return await self._task  # wait until task is completed
         except asyncio.CancelledError:  # ignoring asyncio.CancelledError
             logger.warning("Task %s was stopped unsafely.", repr(self))
+
+
+class UpgradePlan(BaseStep):
+    """Represents the upgrade plan.
+
+    This class is intended to be used as a higher-level group for actual upgrade steps, therefore
+    doesn't accept coroutine or parallel as inputs.
+    """
+
+    prompt: bool = False
+
+    def __init__(
+        self,
+        description: str = "",
+    ):
```

Comment:
not-blocking: Do we expect to have empty `UpgradePlan` or `ApplicationUpgradePlan`, without description?
```suggestion
    def __init__(self, description: str):
```